src/main/kotlin/uk/gov/dluhc/printapi/database/entity/PrintDetails.kt

```diff
@@ -0,0 +1,52 @@
+package uk.gov.dluhc.printapi.database.entity
+
+import software.amazon.awssdk.enhanced.dynamodb.mapper.annotations.DynamoDbBean
+import software.amazon.awssdk.enhanced.dynamodb.mapper.annotations.DynamoDbPartitionKey
+import software.amazon.awssdk.enhanced.dynamodb.mapper.annotations.DynamoDbSecondaryPartitionKey
+import software.amazon.awssdk.enhanced.dynamodb.mapper.annotations.DynamoDbSecondarySortKey
+import software.amazon.awssdk.enhanced.dynamodb.mapper.annotations.DynamoDbSortKey
+import java.time.LocalDate
+import java.time.OffsetDateTime
+import java.util.UUID
+
+@DynamoDbBean
+data class PrintDetails(
+    @get:DynamoDbPartitionKey var id: UUID? = null,
+    @get:DynamoDbSecondaryPartitionKey(indexNames = [REQUEST_ID_INDEX_NAME]) var requestId: String? = null,
+    var sourceReference: String? = null,
+    var applicationReference: String? = null,
+    @get:DynamoDbSecondaryPartitionKey(indexNames = [SOURCE_TYPE_GSS_CODE_INDEX_NAME]) var sourceType: SourceType? = null,
+    var vacNumber: String? = null,
+    var requestDateTime: OffsetDateTime? = null,
+    var firstName: String? = null,
+    var middleNames: String? = null,
+    var surname: String? = null,
+    var certificateLanguage: CertificateLanguage? = null,
+    var photoLocation: String? = null,
+    var delivery: CertificateDelivery? = null,
+    @get:DynamoDbSortKey @get:DynamoDbSecondarySortKey(indexNames = [SOURCE_TYPE_GSS_CODE_INDEX_NAME]) var gssCode: String? = null,
+    var issuingAuthority: String? = null,
+    var issueDate: LocalDate = LocalDate.now(),
+    var eroEnglish: ElectoralRegistrationOffice? = null,
+    var eroWelsh: ElectoralRegistrationOffice? = null
+) {
```

Comment:
Is there no status included in the print details?  
Is there no batchId included in the print details?

If there are, will we need an index to allow finding the print details for a particular status and batch id?

Reply:
@Neil-Massey , I have pushed the changes accordingly.

I added a Global Secondary Index for status being the partition key and batchId being the sortKey.

For the index, @Neil-Massey has one question:
> I wonder if that GSI can be a concatenation of say GSS_CODE#STATUS#BATCH_ID

What do you think, @NathanRussellValtech?

By the way, since we only have partition key and sort key for a global secondary index, seems like we need to do a concatenation if we also include gssCode, and I am not sure, if this is something straightforward to implement.

Reply:
As agreed the Print Batches are not partitioned by GSS Code (ERO) and a batch that is sent to the printer can include a mix of applications across different GSS Codes.  Therefore there is no need to include the GSS Code in the index. 👍